test/SafeProtocolMediator.spec.ts

```diff
@@ -0,0 +1,496 @@
+                    hre.ethers.ZeroAddress,
+                    hre.ethers.ZeroAddress,
+                ]);
+                await expect(safe.exec(await safeProtocolMediator.getAddress(), 0, data))
+                    .to.be.revertedWithCustomError(safeProtocolMediator, "InvalidModuleAddress")
+                    .withArgs(hre.ethers.ZeroAddress);
+            });
+
+            it("Should not allow a Safe to disable SENTINEL_MODULES module", async () => {
+                const { safeProtocolMediator, safe, module } = await loadFixture(deployContractsFixture);
+                await safe.setModule(await safeProtocolMediator.getAddress());
+                const data = safeProtocolMediator.interface.encodeFunctionData("disableModule", [SENTINEL_MODULES, SENTINEL_MODULES]);
+                await expect(safe.exec(await safeProtocolMediator.getAddress(), 0, data))
+                    .to.be.revertedWithCustomError(safeProtocolMediator, "InvalidModuleAddress")
+                    .withArgs(SENTINEL_MODULES);
+            });
+
+            it("Should revert if nexModulePtr and module address do not match", async () => {
+                const { safeProtocolMediator, safe, module } = await loadFixture(deployContractsFixture);
+                await safe.setModule(await safeProtocolMediator.getAddress());
+                const data = safeProtocolMediator.interface.encodeFunctionData("disableModule", [
+                    SENTINEL_MODULES,
+                    await module.getAddress(),
+                ]);
+                await expect(safe.exec(await safeProtocolMediator.getAddress(), 0, data))
+                    .to.be.revertedWithCustomError(safeProtocolMediator, "InvalidPrevModuleAddress")
+                    .withArgs(SENTINEL_MODULES);
+            });
+
+            it("Should disable a module", async () => {
+                const { safeProtocolMediator, safe, module } = await loadFixture(deployContractsFixture);
+                await safe.setModule(await safeProtocolMediator.getAddress());
+                const data = safeProtocolMediator.interface.encodeFunctionData("enableModule", [await module.getAddress(), false]);
+                await safe.exec(await safeProtocolMediator.getAddress(), 0, data);
+                expect(await safeProtocolMediator.getModuleInfo(await safe.getAddress(), await module.getAddress())).to.eql([
+                    false,
+                    SENTINEL_MODULES,
+                ]);
+
+                const data2 = safeProtocolMediator.interface.encodeFunctionData("disableModule", [
+                    SENTINEL_MODULES,
+                    await module.getAddress(),
+                ]);
+                await safe.exec(await safeProtocolMediator.getAddress(), 0, data2);
+                expect(await safeProtocolMediator.getModuleInfo(await safe.getAddress(), await module.getAddress())).to.eql([
+                    false,
+                    ZeroAddress,
+                ]);
+            });
```

Comment:
missing an empty line after the test. we should setup a TS linter and formatter as well

Reply:
we have lint setup in #17 now and formatter is also setup but we dont have them running in pre-commit hooks